### PR TITLE
[BEAM-1902] Retry on correct error codes for datastoreio

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @aaltay 
CC: @eddavisson

File: sdks/python/apache_beam/io/gcp/datastore/v1/helper.py [hunk 1]
```diff
@@ -24,13 +24,13 @@
...
   from google.cloud.proto.datastore.v1 import query_pb2
+  from google.rpc import code_pb2
   from googledatastore import PropertyFilter, CompositeFilter
   from googledatastore import helper as datastore_helper
   from googledatastore.connection import Datastore
   from googledatastore.connection import RPCError
-  QUERY_NOT_FINISHED = query_pb2.QueryResultBatch.NOT_FINISHED
 except ImportError:
-  QUERY_NOT_FINISHED = None
+  pass
```
Comment (aaltay) [inline sdks/python/apache_beam/io/gcp/datastore/v1/helper.py]: @sb2nov are we using `QUERY_NOT_FINISHED` in any place as a check for `gcp` extra package?

Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/datastore/v1/helper.py]: No I think the L224 was the only place and that is gone now.

Comment (aaltay) [inline sdks/python/apache_beam/io/gcp/datastore/v1/helper.py]: Thank you.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**